File: markdown_to_gemini.js
function markdownToGemini() {
  // Geminiの設定 (実際の値に置き換えてください)
  const GEMINI_API_URL = "https://gemini.example.com/v1/import";
  const GEMINI_API_KEY = "YOUR_GEMINI_API_KEY";
  const INPUT_DIR = "markdown_files"; // Google Driveのフォルダ名

  function readMarkdownFiles() {
    const documents = [];
    // Google Driveのフォルダを取得
    let folder = DriveApp.getRootFolder().getFoldersByName(INPUT_DIR);
    if (folder.hasNext()) {
      folder = folder.next();
    } else {
      Logger.log("Markdownファイルが見つかりませんでした。");
      return [];
    }

    const files = folder.getFilesByType(MimeType.PLAIN_TEXT);
    while (files.hasNext()) {
      const file = files.next();
      if (file.getName().endsWith(".md")) {
        const content = file.getBlob().getDataAsString();
        const title = content.split("
")[0].replace("# ", "");
        documents.push({
          "title": title,
          "content": content,
          "metadata": { "source": "Confluence" }
        });
      }
    }
    return documents;
  }

  function sendToGemini(documents) {
    const headers = {
      "Authorization": `Bearer ${GEMINI_API_KEY}`,
      "Content-Type": "application/json"
    };
    const options = {
      "method": "post",
      "headers": headers,
      "payload": JSON.stringify({ "documents": documents }),
      "muteHttpExceptions": true
    };
    const response = UrlFetchApp.fetch(GEMINI_API_URL, options);
    if (response.getResponseCode() !== 200) {
      Logger.log(`Error sending to Gemini: ${response.getContentText()}`);
      return null;
    }
    return JSON.parse(response.getContentText());
  }

  // スクリプトの実行
  const documents = readMarkdownFiles();
  if (documents.length > 0) {
    const result = sendToGemini(documents);
    Logger.log("Geminiへの送信結果:", result);
  } else {
    Logger.log("Markdownファイルが見つかりませんでした。");
  }
}

File: markdown_to_gemini.py
import os
import json
import requests

# Geminiの設定
GEMINI_API_URL = "https://gemini.example.com/v1/import"  # 実際のAPIエンドポイント
GEMINI_API_KEY = "YOUR_GEMINI_API_KEY"

# ファイルのディレクトリ
INPUT_DIR = "./markdown_files"


def read_markdown_files():
    """Markdownファイルを読み込み、Gemini用のJSON形式に変換"""
    documents = []
    for filename in os.listdir(INPUT_DIR):
        if filename.endswith(".md"):
            with open(os.path.join(INPUT_DIR, filename), "r") as f:
                content = f.read()
                title = content.split("\n")[0].replace(
                    "# ", "")  # 最初の見出しをタイトルとして使用
                documents.append({
                    "title": title,
                    "content": content,
                    "metadata": {"source": "Confluence"}
                })
    return documents


def send_to_gemini(documents):
    """Gemini APIにデータを送信"""
    headers = {
        "Authorization": f"Bearer {GEMINI_API_KEY}",
        "Content-Type": "application/json"
    }
    response = requests.post(
        GEMINI_API_URL,
        headers=headers,
        json={"documents": documents}
    )
    return response.json()


if __name__ == "__main__":
    documents = read_markdown_files()
    if documents:
        result = send_to_gemini(documents)
        print("Geminiへの送信結果:", result)
    else:
        print("Markdownファイルが見つかりませんでした。")
